kapitan/refs/cmd_parser.py

```diff
@@ -115,6 +116,28 @@ def ref_write(args, ref_controller):
         tag = "?{{awskms:{}}}".format(token_path)
         ref_controller[tag] = secret_obj
 
+    elif token_name.startswith("azkms:"):
+        type_name, token_path = token_name.split(":")
+        key = args.key
+        if args.target_name:
+            inv = inventory_reclass(args.inventory_path)
+            kap_inv_params = inv["nodes"][args.target_name]["parameters"]["kapitan"]
+            if "secrets" not in kap_inv_params:
+                raise KapitanError(
+                    "parameters.kapitan.secrets not defined in inventory of target {}".format(
+                        args.target_name
+                    )
+                )
+
+            key = kap_inv_params["secrets"]["azkms"]["key"]
```

Comment:
This will raise TypeError/KeyError if either `parameters.kapitan.secrets.azkms` or `parameters.kapitan.secrets.azkms.key` 
We should either implement jsonschema validation for this parameter structure or just turn it into something like 
```python
try:
  key = kap_inv_params["secrets"]["azkms"]
  except (KeyError, TypeError):
  raise KapitanError(
                    "parameters.kapitan.secrets.azkms.key not defined in inventory of target {}".format(
                        args.target_name
                    )
```
And I just realised we have the same issue for gpg/awskms/gkms...